### ISSUE TITLE
`{{@value}}` prefers to use a helper named `value` instead of a data variable named `value`

### ISSUE THREAD

**Issue by pepkin88** [wycats/handlebars.js#1465]
I'm currently upgrading from Handlebars v1 to v4 in my project and I stumbled upon this strange behavior, which seems like a bug to me.

Test case: https://jsfiddle.net/64zfydjp/
_(shows `helper helper` instead of `helper data`)_

Previous behavior (last version: 2.0.0): https://jsfiddle.net/9rtn75sq/2/
_(shows `helper data`, as I expected)_

If I'm using `@` I'm expecting to use a data variable, not a helper. But even I wanted a helper, it should be named `@value`, not just `value`. Moreover, the helper is invoked identically, so I cannot differentiate those cases in the helper code.

Is this behavior really expected or is it a bug? If it's expected, how could I differentiate `{{@value}}` from `{{value}}`?

**Comment by nknapp** [comment]
I have to check this. Seems to me like a bug, but you never know until you see the change that's causing it and the reason why it was introduced.

**Comment by nknapp** [comment]
I believe that commit 884bf1553663734f22ffcd9d758c9d71d4373bf9 introduced this behavior. I have not tested it thought. 

The precompiler compiles `{{@value}}` and `{{value}}` to the same code. This is certainly strange. But it has been in the code since 3.0.0 and I would be very hesitant to change it. I can already smell unexpected side effects from changing it (and bug reports).

I may change my mind on this, but it's certainly not something that is fixed quickly.

**Comment by pepkin88** [comment]
The hardest part for me, was to locate the reason why my helper stopped working correctly. After that I applied a fix/workaround that was enough for my project. That means there are still some edge cases which would cause a collision, but now I'm aware of them.

So I can live without a proper fix for it. And maybe this Issue will help someone else with a similar problem.

If someone was interested about my use case, I made a block helper named `v`, which executes a block if a passed value was truthy, with assigning that value to a variable `@v` (easy to remember). Example usage:
```hbs
{{#v user.fullName}}Name: {{@v}}{{/v}}
```
